autometa/common/external/diamond.py

```diff
@@ -157,66 +269,72 @@ def blast(
+        '</path/to/diamond/formatted/database>'
     outfpath : str
-        </path/to/output/file>.
-    blast_type : str
-        blastp if fasta consists of amino acids or blastx if nucleotides
-    evalue : float
-        cutoff e-value to count hit as significant (the default is float('1e-5')).
-    maxtargetseqs : int
-        max number of target sequences to retrieve per query by diamond (the default is 200).
-    cpus : int
-        number of cpus to use (the default is 1).
-    tmpdir : type
-        </path/to/temporary/directory> (the default is os.curdir).
-    force : boolean
-        overwrite existing diamond results `force` (the default is False).
-    verbose : boolean
-        log progress to terminal `verbose` (the default is False).
+        Path to output file
+        '</path/to/output/file>'
+    blast_type : str, optional
+        blastp to align protein query sequences against a protein reference database,
+        blastx to align translated DNA query sequences against a protein reference database, by default 'blastp'
+    evalue : float, optional
+        cutoff e-value to count hit as significant, by default float('1e-5')
+    maxtargetseqs : int, optional
+        max number of target sequences to retrieve per query by diamond, by default 200
+    cpus : int, optional
+        Number of processors to be used, by default uses all the processors of the system
+    tmpdir : str, optional
+        Path to temporary directory
+        '</path/to/temporary/directory>' by default tempfile.mkdtemp()
+    force : bool, optional
+        overwrite existing diamond results, by default False
+    verbose : bool, optional
+        log progress to terminal, by default False
 
     Returns
     -------
     str
-        `outfpath`
+        Path to BLAST results
 
     Raises
-    -------
+    ------
     FileNotFoundError
         `fasta` file does not exist
     ValueError
         provided `blast_type` is not 'blastp' or 'blastx'
-    OSError
-        Diamond execution failed
+    subprocess.CalledProcessError
+        Failed to run blast
     """
     if not os.path.exists(fasta):
         raise FileNotFoundError(fasta)
     if os.path.exists(outfpath) and not force:
-        empty = not os.stat(outfpath).st_size
-        if not empty:
-            if verbose:
-                logger.warning(f"FileExistsError: {outfpath}. To overwrite use --force")
+        if os.path.getsize(outfpath) and verbose:
+            logger.warning(f"FileExistsError: {outfpath}. To overwrite use --force")
             return outfpath
     blast_type = blast_type.lower()
     if blast_type not in ["blastp", "blastx"]:
         raise ValueError(f"blast_type must be blastp or blastx. Given: {blast_type}")
+    if tmpdir == None:
+        tmpdir = tempfile.mkdtemp()
     if verbose:
-        logger.debug(f"Diamond{blast_type.title()} {fasta} against {database}")
+        logger.debug(f"Diamond{blast_type} {fasta} against {database}")
```

Comment:
minor changes for whitespace and executable

```suggestion
        logger.debug(f"diamond {blast_type} {fasta} against {database}")
```